crates/cli/src/subcommands/login.rs

```diff
@@ -167,3 +186,19 @@ async fn spacetimedb_login(remote: &str, web_session_id: &String) -> Result<Stri
 
     Ok(response.token.clone())
 }
+
+#[derive(Debug, Clone, Deserialize)]
+struct LocalLoginResponse {
+    pub token: String,
+}
+
+async fn spacetimedb_direct_login(host: &str) -> Result<String, anyhow::Error> {
+    // Users like to provide URLs with trailing slashes, which can cause issues due to double-slashes in the routes below.
+    let host = host.trim_end_matches('/');
```

Comment:
I believe there is a way with `PathBuf` to just have this do this for you. https://doc.rust-lang.org/std/path/struct.Path.html#method.join I believe this merges trailing slashes.

Reply:
I don't believe pathbuf is suitable for URLs unless we keep the protocol separate until the very end (`http://foo.com` is parsed differently as a path than as a url)